App Details Panel and Cards

denysbutenko opened the pull request:
You can check the new apps page and select any installed apps for testing.

bugclerk commented:
Jira URL: https://ixsystems.atlassian.net/browse/NAS-120088

dszidi reviewed (commented):
Off to a good start. Cards are present and update on table row click as expected.
The application card is missing the Admin Portal button and the Update button but otherwise looks ok.

The Containers card looks good but the shell action needs to be moved. We don't invoke a shell for the app as each app can have multiple containers. For this reason we should remove the "Shell" button from the card header and provide one for each individual pod list items. I can adjust the existing mockup to reflect that change if need be.

The History card looks good and I'm assuming the endpoint for Notes is not ready yet? Those should be provided by middleware

denysbutenko commented:
> Off to a good start. Cards are present and update on table row click as expected. The application card is missing the Admin Portal button and the Update button but otherwise looks ok.
 
The `Web/Admin Portal` buttons are coming from the `app.portals` property and appear dynamically. 
The `Update` button relies on `app.update_available` property. I going to sync the condition with the old apps module. Thanks for pointing that out.

> The Containers card looks good but the shell action needs to be moved. We don't invoke a shell for the app as each app can have multiple containers. For this reason we should remove the "Shell" button from the card header and provide one for each individual pod list items. I can adjust the existing mockup to reflect that change if need be.

It makes sense, agreed.

> The History card looks good and I'm assuming the endpoint for Notes is not ready yet? Those should be provided by middleware

Yeah, I'll replace static content when new data is available.

denysbutenko commented:
@dszidi ok, I've addressed the feedback. The `Shell` and `View Logs` buttons will appear as icon buttons with a tooltip. Although the dialogs require further work to accept pods outside, we will make sure to update them at a later time. I am open to suggestions about changing icons or the approach itself if needed.

Preview of changes:
![image](https://user-images.githubusercontent.com/351613/225847312-5177b9fe-b738-4f38-843c-733b67ea8151.png)

denysbutenko commented:
> We should remove the underlines from the icons
> 
> <img alt="image" width="135" src="https://user-images.githubusercontent.com/29865743/226401170-88c53b29-362a-4408-bb79-bb7f7845fe78.png">

Oh yeah, it was removed.

> > The Update button relies on app.update_available property. I going to sync the condition with the old apps module.
> 
> Will this be done for this PR or as a separate ticket item? Currently the update button is still missing

The condition was updated to include both parameters - `update_available` and `container_images_update_available`. Perhaps, I misunderstood the requirements. The initial assumption was to show the button when an update is available and hide it when there are no updates. However, I made changes to make the button visible all time and disabled it when there were no updates.

dszidi reviewed (changes requested):
> ... Perhaps, I misunderstood the requirements. The initial assumption was to show the button when an update is available and hide it when there are no updates. However, I made changes to make the button visible all time and disabled it when there were no updates.

Apologies. You didn't actually misunderstand the requirements. While testing I mistakenly thought an update was available for the app I was looking at and the button wasn't showing. On closer inspection it turns out there wasn't an update and your button was doing the right thing. No need to show the update button if there isn't one available.

Beyond that, the rest looks good to me

<img width="945" alt="image" src="https://user-images.githubusercontent.com/29865743/227222716-13b7d90f-e456-4437-b9bf-a23fc9c1d40a.png">

denysbutenko commented:
> > ... Perhaps, I misunderstood the requirements. The initial assumption was to show the button when an update is available and hide it when there are no updates. However, I made changes to make the button visible all time and disabled it when there were no updates.
> 
> Apologies. You didn't actually misunderstand the requirements. While testing I mistakenly thought an update was available for the app I was looking at and the button wasn't showing. On closer inspection it turns out there wasn't an update and your button was doing the right thing. No need to show the update button if there isn't one available.
> 
> Beyond that, the rest looks good to me
> 
> <img alt="image" width="945" src="https://user-images.githubusercontent.com/29865743/227222716-13b7d90f-e456-4437-b9bf-a23fc9c1d40a.png">

Ok, no problem. Fixed that.

dszidi reviewed (commented):
After having updated Plex using bulk actions, the app details card still shows the update button. When I press it gives an error dialog

<img width="1789" alt="image" src="https://user-images.githubusercontent.com/29865743/227810205-12f821a3-635d-4fec-8165-91ef85d69b2d.png">

denysbutenko commented:
> After having updated Plex using bulk actions, the app details card still shows the update button. When I press it gives an error dialog
> 
> <img alt="image" width="1789" src="https://user-images.githubusercontent.com/29865743/227810205-12f821a3-635d-4fec-8165-91ef85d69b2d.png">

Good catch, @dszidi. The dialog invokes refreshing the data on close, but catching the finishing update event is tricky. It should not be an issue when the app data store is implemented. I'll keep that in mind for future improvements.

dszidi reviewed (approved):
LGTM
